packages/backend/src/app.ts

```diff
@@ -393,21 +395,52 @@ export class App {
     )
 
     let tenantApiSignatureResult: TenantApiSignatureResult
-    if (this.config.env !== 'test') {
-      koa.use(async (ctx, next: Koa.Next): Promise<void> => {
-        const result = await getTenantFromApiSignature(ctx, this.config)
-        if (!result) {
-          ctx.throw(401, 'Unauthorized')
-        } else {
+    const tenantSignatureMiddleware = async (
+      ctx: AppContext,
+      next: Koa.Next
+    ): Promise<void> => {
+      const result = await getTenantFromApiSignature(ctx, this.config)
+      if (!result) {
+        ctx.throw(401, 'Unauthorized')
+      } else {
+        tenantApiSignatureResult = {
+          tenant: result.tenant,
+          isOperator: result.isOperator ? true : false
+        }
+      }
+      return next()
+    }
+
+    const testTenantSignatureMiddleware = async (
+      ctx: AppContext,
+      next: Koa.Next
+    ): Promise<void> => {
+      if (ctx.headers['tenant-id']) {
+        const tenantService = await ctx.container.use('tenantService')
+        const tenant = await tenantService.get(
+          ctx.headers['tenant-id'] as string
+        )
+
+        if (tenant) {
           tenantApiSignatureResult = {
-            tenant: result.tenant,
-            isOperator: result.isOperator ? true : false
+            tenant,
+            isOperator: tenant.apiSecret === this.config.adminApiSecret
           }
+        } else {
+          ctx.throw(401, 'Unauthorized')
         }
-        return next()
-      })
+      }
+      return next()
     }
 
+    // For tests, we still need to get the tenant in the middleware, but
+    // we don't need to verify the signature nor prevent replay attacks
+    koa.use(
+      this.config.env !== 'test'
+        ? tenantSignatureMiddleware
+        : testTenantSignatureMiddleware
+    )
```

Comment:
Basically, had an issue where the replay attack prevention (ie `canApiSignature` function) was resulting a 401 Unauthorized error for many resolver tests (for cases where we call the same resolver back to back)